src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Refactoring/CodeActionResolutionEndpointTest.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using Microsoft.AspNetCore.Razor.LanguageServer.Refactoring;
+using Microsoft.AspNetCore.Razor.LanguageServer.ProjectSystem;
+using Microsoft.AspNetCore.Razor.Test.Common;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Editor.Razor;
+using Moq;
+using OmniSharp.Extensions.LanguageServer.Protocol.Client.Capabilities;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using Xunit;
+using System;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+using System.Linq;
+using System.Threading;
+using System.Collections.Generic;
+using Newtonsoft.Json.Linq;
+using System.Diagnostics;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Test.Refactoring
+{
+    public class CodeActionResolutionEndpointTest : LanguageServerTestBase
+    {
+        private readonly DocumentResolver EmptyDocumentResolver = Mock.Of<DocumentResolver>();
+
+        [Fact]
+        public async Task Handle_Resolve()
+        {
+            // Arrange
+            var codeActionEndpoint = new CodeActionResolutionEndpoint(new RazorCodeActionResolver[] {
+                new MockCodeActionResolver("Test"),
+            }, LoggerFactory);
+            var request = new RazorCodeActionResolutionParams()
+            {
+                Action = "Test",
+                Data = null
+            };
+
+            // Act
+            var workspaceEdit = await codeActionEndpoint.Handle(request, default);
+
+            // Assert
+            Assert.NotNull(workspaceEdit);
```

Comment:
I usually like it assert a couple things about the expected contents of the returned object to ensure it got constructed properly.

Reply:
Normally I would agree but in this case it's just a mock endpoint returning a `new WorkspaceEdit()`. No fields to check!